plugins/woocommerce/client/legacy/js/admin/wc-admin-menu.js

```diff
@@ -0,0 +1,20 @@
+document.addEventListener( 'DOMContentLoaded', () => {
+
+	const adminMenu = document.getElementById( 'adminmenu' );
+	if ( ! adminMenu ) {
+		return;
+	}
+
+	const addonsSubMenuLinks = adminMenu.querySelectorAll(
+		'#toplevel_page_woocommerce' +
+		' .wp-submenu' +
+		' li a[href="admin.php?page=wc-addons"]'
+	);
```

Comment:
Not tested, but wouldn't this be more-concise:

```suggestion
	const addonsSubMenuLinks = document.querySelectorAll(
		'#adminmenu' +
		' #toplevel_page_woocommerce' +
		' .wp-submenu' +
		' li a[href="admin.php?page=wc-addons"]'
	);
```

It'd require one fewer DOM lookup in the (common) case that the admin menu is present, and in the rare event that there's no admin menu visible (I don't even know how or when that'd happen!), the only "cost" is the use of a `querySelectorAll` to produce an empty `NodeList` rather than a `getElementById` to produce a `null` response.

Your approach is arguably "right" because `getElementById` _should_ be more-performant than `querySelectorAll` (thanks to a standard kicked-off by Internet Explorer of all things, IDs of first/unique elements on a page get precached into a lookup table when the DOM is rendered, making lookups quick). But the real impact is marginal and, again, **the benefit of your approach only exists on admin pages where there is no admin menu** (which might well be _none_ of them)!

Not a big deal either way though. If you find your way more-readable, keep it!

Reply:
Thanks @Dan-Q, I was indeed trying to reduce unnecessary lookups with the first bit. Also, once we've found `#adminmenu`, the second selection is cheaper than a full document search. But presumably, a `querySelectorAll` starting with an ID selector uses the lookup table to start with, and as you suggest, we're guarding against a highly unlikely situation. 

We're also nesting ID selectors, which shouldn't really be necessary – but I wanted to make sure in case `#toplevel_page_woocommerce` is used in some other context.